crowbar_framework/app/models/service_object.rb

```diff
@@ -978,7 +997,17 @@ def apply_role(role, inst, in_queue, bootstrap = false)
     else
       # Attempt to queue the proposal.  If delay is empty, then run it.
       deps = proposal_dependencies(role)
-      delay, pre_cached_nodes = queue_proposal(inst, element_order, new_elements, deps)
+      if skip_unready_nodes_enabled
+        elements_without_unready, pre_cached_nodes = skip_unready_nodes(
+          @bc_name, inst, new_elements, old_elements
+        )
+        delay, pre_cached_nodes = queue_proposal(
+          inst, element_order, elements_without_unready, deps, pre_cached_nodes
```

Comment:
Missing `@bc_name` as parameter, we are now passing `pre_cached_nodes` as `@bc_name` so the queue function gets craaaazy